uobtheatre/bookings/test/test_models.py

```diff
@@ -877,37 +877,53 @@ def test_booking_pay_deletes_pending_payments():
 
 @pytest.mark.django_db
 @pytest.mark.parametrize(
-    "initial_state, final_state",
-    [(True, True), (False, True)],
+    "initial_check_time, initial_check_state, final_check_state",
+    [
+        (None, False, True),
+        (datetime.datetime.fromisoformat("2000-01-01T00:00:00+00:00"), True, True),
+    ],
 )
-def test_ticket_check_in(initial_state, final_state):
+def test_ticket_check_in(
+    gql_client, initial_check_time, initial_check_state, final_check_state
+):
     """
     Test ticket check in method
     """
+    mock_ticket_check_in_time = datetime.datetime(2020, 1, 2, 23, 45)
+    with patch.object(timezone, "now", return_value=mock_ticket_check_in_time) as _:
+
+        gql_client.login()
+
+        ticket = TicketFactory(checked_in_at=initial_check_time)
 
-    ticket = TicketFactory(checked_in=initial_state)
+        assert ticket.checked_in == initial_check_state
 
-    assert ticket.checked_in == initial_state
-    ticket.check_in()
-    assert ticket.checked_in == final_state
+        ticket.check_in(user=gql_client.user)
+        assert ticket.checked_in == final_check_state
+        assert ticket.checked_in_at == mock_ticket_check_in_time
+        assert ticket.checked_in_by == gql_client.user
```

Comment:
Not quite sure why you are using `gql_client` in a model unit test - it looks like just to get a user?
Just use `user = UserFactory()` instead.